main.tf

```
@@ -19,7 +19,8 @@ locals {
 }
 
 data "github_repository" "default" {
-  name = try(github_repository.default.0.name, var.name)
+  name       = try(github_repository.default.0.name, var.name)
+  depends_on = [github_repository.default]
```

Before it was this, should that work? I guess so:
```suggestion
  name = var.create_repository ? github_repository.default.0.name : var.name
```